activerecord/lib/active_record/timestamp.rb

```diff
@@ -107,7 +107,7 @@ def all_timestamp_attributes
 
     def max_updated_column_timestamp(timestamp_names = timestamp_attributes_for_update)
       timestamp_names
-        .map { |attr| self[attr] }
+        .map { |attr| self[attr] || self.respond_to?(attr) }
```

Comment:
won't `respond_to?` just return true or false rather than actual timestamp? perhaps you need `try(attr)` instead